src/libraries/System.Private.CoreLib/src/System/IO/BinaryReader.cs

```diff
@@ -472,32 +471,30 @@ public virtual byte[] ReadBytes(int count)
 
             if (numRead != result.Length)
             {
-                // Trim array.  This should happen on EOF & possibly net streams.
-                byte[] copy = new byte[numRead];
-                Buffer.BlockCopy(result, 0, copy, 0, numRead);
-                result = copy;
+                // Trim array. This should happen on EOF & possibly net streams.
+                result = result.AsSpan()[..numRead].ToArray();
```

Comment:
```suggestion
                result = result[..numRead];
```

Reply:
Thought it was less efficient but [it wasn't](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACY8pJ0hgYQYG8aH+B/ekxYNgATwwwA2gF0GAJRjYAJgCFJMXAAoAlgDsMDSAFdDaBgaOnDpAJR9BA3tSdvLAMwbabRgLx+DAAMDq7ugi7h4cQA7AwAglBQ2OIAdACi+AAOGOIAPBJSAHzadgDcjlEMAL40lVGFMvKwuCYANv4M+jAA7mKa0r6yFWFVul4+EGYY7ACEgS3tGKkAMjD6AOYYABahVfyR+4IA9McMACpQuvgM2EkpqRfburgMuNtTbSoM29hZWesGBB9Ax0gB5ABiDAAZAwshBcLhdMA2uIujAjLgMLBsPhcKl6kdFh0GAstEtUvFcABlLLYfSlaSpVK+UiyVLnCCJZLiUojI61aiE9yxBjEjD89yC6KMZisRpyRTKdSaXCkPSGYxTcyWTWsvbuQ5jCa+UmBELCtxGo5MOLch6ZHL5RolcqWpyC92CBXNckkwLdPoKoaS43eVkMeZiv3LNabHYGqrWo6nC5XG53HmPc7PV7vT7fX7/QHA0GQmFwhFIlFo7qY7HKPEE0Y28Wk6OtDpMlnambDL0CT0t6JxcWhj00apAA==), thanks!